Cannot compile on 32-bit platform

The following line causes a compilation error: https://github.com/cberner/redb/blob/7762a6e6f65a7823e6780ae99fbbb799a7374b82/src/tree_store/page_store/page_manager.rs#L49

```
error[E0080]: erroneous constant used
   --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/redb-0.3.1/src/tree_store/page_store/page_manager.rs:461:63
    |
461 |             let region_size = requested_region_size.unwrap_or(MAX_USABLE_REGION_SPACE);
    |                                                               ^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors
```

`usize` on this platform is 32-bits, so there's an overflow by 1. A naive fix is to change this type to u64, but I'm unsure if this has further implications, so I wanted to raise the issue. 

Generally, is redb suitable for 32-bit platforms?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -34,7 +34,7 @@ jobs:
         if: startsWith(matrix.os, 'ubuntu')
         run: |
           sudo apt update
-          sudo apt install -y python3 python3-pip python3-dev python3-venv
+          sudo apt install -y python3 python3-pip python3-dev python3-venv gcc-multilib
 
       - name: Install packages
         if: startsWith(matrix.os, 'macos')
@@ -44,14 +44,19 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
+          target: i686-unknown-linux-gnu
           components: rustfmt, clippy
 
       - name: Install cargo-deny
         if: steps.rust-cache.outputs.cache-hit != 'true'
         run: cargo install --force --version 0.11.4 cargo-deny --locked
 
-      - name: Compile
+      - name: Compile x64
         run: cargo build --all-targets --all-features
+
+      - name: Compile 32bit
+        if: startsWith(matrix.os, 'ubuntu')
+        run: cargo build --target i686-unknown-linux-gnu
 
       - name: Run tests
         run: make build test

--- a/src/tree_store/page_store/page_manager.rs
+++ b/src/tree_store/page_store/page_manager.rs
@@ -46,7 +46,11 @@ use std::sync::{Mutex, MutexGuard};
 
 // Regions have a maximum size of 4GiB. A `4GiB - overhead` value is the largest that can be represented,
 // because the leaf node format uses 32bit offsets
+#[cfg(target_pointer_width = "64")]
 const MAX_USABLE_REGION_SPACE: usize = 4 * 1024 * 1024 * 1024;
+// Default to only 128MiB on 32bit platforms, since the mmap likely cannot be 4GiB or even close.
+#[cfg(target_pointer_width = "32")]
+const MAX_USABLE_REGION_SPACE: usize = 128 * 1024 * 1024;
 pub(crate) const MAX_PAGE_ORDER: usize = 20;
 pub(super) const MIN_USABLE_PAGES: usize = 10;
 const MIN_DESIRED_USABLE_BYTES: usize = 1024 * 1024;

